csrc/alias_analysis.cpp

```diff
@@ -248,6 +248,11 @@ void AliasFinder::handle(const LoadStoreOp* set) {
   }
   TensorView* out = set->out()->as<TensorView>();
 
+  // The set operation is an inplace update and shouldn't be marked as alias op.
+  if (out->fusion()->getOutputAlias(out).type == AllocationType::ReuseBuffer) {
```

Comment:
We may want to check this in AliasAnalysisResult::finalize. Keeping this alias pair in `alias_to_source_` is useful if the result of the `set` is also used by other aliasing ops. I believe as long as we don't materialize it in `alias_to_root_` we should be fine. 

https://github.com/NVIDIA/Fuser/blob/e7d551e908fd691e791bc34a966c80bdda34f36b/csrc/alias_analysis.cpp#L452

Reply:
Good point. Let me take another look at this later.